### PR TITLE
Adds pin reset flow

### PR DESCRIPTION
**Pull request by mango-habanero**

### CODE REVIEW

File: app/server/models/user.py [hunk 1]
```diff
@@ -560,6 +560,11 @@ def check_reset_token_already_used(self, password_reset_token):
         is_valid = password_reset_token in self.password_reset_tokens
         return is_valid
 
+    def is_pin_reset_token_valid(self, pin_reset_token):
+        self.clear_expired_pin_reset_tokens()
+        pin_reset_token_in_valid_reset_tokens = pin_reset_token in self.pin_reset_tokens
```
Comment (nolash) [inline app/server/models/user.py:565]: I find it more readable if you directly return the evaluation

File: app/server/utils/user.py [hunk 2]
```diff
@@ -850,7 +850,7 @@ def create_user_without_transfer_account(phone):
                 phone=phone,
                 registration_method=RegistrationMethodEnum.USSD_SIGNUP)
 
-    organisation = g.active_organisation
+    organisation = Organisation.query.get(2)
```
Comment (nolash) [inline app/server/utils/user.py:855]: Why is this necessary?

File: app/server/utils/ussd/kenya_ussd_state_machine.py [hunk 3]
```diff
@@ -422,6 +422,15 @@ def has_complete_profile(self, user_input):
         else:
             return False
 
+    def is_admin_pin_reset(self, user_input):
+        pin_reset_tokens = self.user.pin_reset_tokens
+        valid_reset_tokens = []
+        for reset_token in pin_reset_tokens:
+            if self.user.is_pin_reset_token_valid(reset_token):
+                valid_reset_tokens.append(reset_token)
```
Comment (nolash) [inline app/server/utils/ussd/kenya_ussd_state_machine.py:431]: This can break here and function just return bool?

File: app/server/models/user.py [hunk 4]
```diff
@@ -560,6 +560,11 @@ def check_reset_token_already_used(self, password_reset_token):
         is_valid = password_reset_token in self.password_reset_tokens
         return is_valid
 
+    def is_pin_reset_token_valid(self, pin_reset_token):
+        self.clear_expired_pin_reset_tokens()
```
Comment (nolash) [inline app/server/models/user.py:564]: do we need to clear all remaining tokens afterwards too, I guess we don't need any once PIN is reset.

**Comment by nolash** [comment]
Don't we need test code for this?

File: app/server/utils/user.py (hunk 2 above)
Comment (mango-habanero) [reply app/server/utils/user.py:855]: I keep forgetting to clean this out, I find myself having to do this because of the fact that I cannot create a user using the external_auth process because its actually broken in the local environment. It really pisses me off.
I shall revert this though

File: app/server/models/user.py (hunk 4 above)
Comment (mango-habanero) [reply app/server/models/user.py:564]: I looked at the DB post reset and the tokens were cleared, not sure where the function for that is and I'll confirm just to be safe but it is there.

File: app/server/utils/user.py (hunk 2 above)
Comment (nolash) [reply app/server/utils/user.py:855]: The organisation stuff is a fucking pain in the ass.

Comment (mango-habanero) [reply app/server/utils/user.py:855]: Yeah. It really is irksome

**Review by nolash — APPROVED** [review]
